### PR TITLE
FIX: Tabs->Spaces and QStrings

### PR DESCRIPTION
**Pull request by Eric89GXL**
This does a few things:

1. Updates the binary. I built it on 14.10 so I need to double-check it works on 14.04.

2. Changes all tabs to spaces.

3. Changes out some `char *` instances for `QString` to move toward more simplicity / consistency.

4. Added a "Limit display size" check box that will be useful for us when ensuring things are properly focused. It makes it so that the display size won't exceed the incoming video size when checked.

5. Makes the videos centered vertically and horizontally (used to be left-justified) in the window.

6. Modifies `.gitignore` to add the build files (`.o`, `ui_*.h`, and `moc_*.cpp`).

7. Modifies `SyncDaemon` to have `-O2` compilation and adds convenience `Makefile`.

8. Updates `pvm_export_dragdrop.py` to use `mencoder` for both steps instead of `avconv` for the -> mov conversion.

I hope I'm not committing too many changes here. Again, feel free to provide feedback about changes if these are not what you want. It's tough to see the substantive changes in the midst of all the tab changes (sorry), so I'll comment in-line where they occurred.

### CODE REVIEW

File: VideoRecStation/src/videodialog.cpp [hunk 1]
```diff
@@ -186,18 +177,15 @@ void VideoDialog::onNewFrame(unsigned char* _jpegBuf)
 {
     ChunkAttrib chunkAttrib;
     float       fps;
-    char        fpsLabelBuff[100];
```
Comment (Eric89GXL) [inline VideoRecStation/src/videodialog.cpp]: This was removed and replaced with a QString below

File: VideoRecStation/src/speakerthread.cpp [hunk 2]
```diff
@@ -26,36 +26,36 @@ using namespace std;
...
 
-	// Open PCM device for playback
-	rc = snd_pcm_open(&sndHandle, settings.outAudioDev, SND_PCM_STREAM_PLAYBACK, 0);
-	if (rc < 0)
-	{
-		cerr << "unable to open pcm device: " << snd_strerror(rc) << endl;
-	    exit(EXIT_FAILURE);
-	}
+    // Open PCM device for playback
+    rc = snd_pcm_open(&sndHandle, settings.outAudioDev.toLocal8Bit().data(), SND_PCM_STREAM_PLAYBACK, 0);
```
Comment (Eric89GXL) [inline VideoRecStation/src/speakerthread.cpp]: `settings.outAudioDev`, `settings.inAudioDev`, and `settings.storagepath` were all replaced with QString instances, so I now change them over to the local8Bit when necessary.

File: VideoRecStation/src/videodialog.cpp [hunk 3]
```diff
@@ -29,42 +29,37 @@ using namespace std;
...
+    Settings    settings;
     cameraIdx = _cameraIdx;
 
-	ui.setupUi(this);
-	setWindowFlags(Qt::Window | Qt::CustomizeWindowHint | Qt::WindowTitleHint| Qt::WindowSystemMenuHint | Qt::WindowMinMaxButtonsHint);
-    sprintf(winCaption, "Camera %i", cameraIdx + 1);
-	setWindowTitle(winCaption);
-	camera = _camera;
+    ui.setupUi(this);
+    ui.videoWidget->setAlignment(Qt::AlignHCenter | Qt::AlignVCenter);
```
Comment (Eric89GXL) [inline VideoRecStation/src/videodialog.cpp]: This sets the widget alignment

File: VideoRecStation/src/maindialog.cpp [hunk 4]
```diff
@@ -148,6 +149,7 @@ void MainDialog::cleanVideoDialog(unsigned int idx)
     settings.videoGains[idx] = videoDialogs[idx]->findChild<QSlider*>("gainSlider")->value();
     settings.videoUVs[idx] = videoDialogs[idx]->findChild<QSlider*>("uvSlider")->value();
     settings.videoVRs[idx] = videoDialogs[idx]->findChild<QSlider*>("vrSlider")->value();
+    settings.videoLimits[idx] = videoDialogs[idx]->findChild<QCheckBox*>("ldsBox")->isChecked();
```
Comment (Eric89GXL) [inline VideoRecStation/src/maindialog.cpp]: You can search for instances of `ldsBox` to find the lines that were changed to add that option.

**Comment by Eric89GXL** [comment]
FYI I used this to change the spaces:
```
find . -name '*.cpp' ! -type d -exec bash -c 'expand -t 4 "$0" > /tmp/e && mv /tmp/e "$0"' {} \;
find . -name '*.h' ! -type d -exec bash -c 'expand -t 4 "$0" > /tmp/e && mv /tmp/e "$0"' {} \;
```
This PR also modifies `.gitignore` to add the build files (`.o`, `ui_*.h`, and `moc_*.cpp`).

**Comment by Eric89GXL** [comment]
Latest commit adds a `Makefile` for SyncDaemon, does `-O2` optimizations on it, and changes tabs->spaces for that file. Is that okay with you?

**Comment by andreyzhd** [comment]
Looks OK. I've also rebuilt the executables.

**Comment by Eric89GXL** [comment]
Did you try the version that I had re-built? It worked on my VideoMEG setup. It would be nice if my builds worked for you, too, so you didn't have to build and push them after any commit to the C code.

**Comment by andreyzhd** [comment]
Your binaries seem to work OK. I rebuilt the binaries because I was not sure whether you did it. One reason not to push the binaries to GitHub is to make merges a little easier (since Git can merge source code much better than binaries). However, it is not such a big difference.
Should we decide that you always put the updated binaries in every PR?

**Comment by Eric89GXL** [comment]
I think it makes sense to upload a new binary with each PR, at least for a while. This has the unfortunate side effect of making the repo grow much faster than it would with just source code changes, though. For this reason, generally the repos I know do not include compiled binaries, instead releasing them through other venues. It's really convenient for our use case to have the binaries with the source, though, so my vote is to stick with that for now. Eventually if the repo gets too big, it is possible to purge old binaries from the repo using `git`. It's a pain, but it's doable.

**Comment by andreyzhd** [comment]
Fine with me.

**Pull request merged**